### PR TITLE
Interface change of StandardDetector and Standard Controller

### LINKED ISSUE #405: Decide the interface between `StandardDetector` and `DetectorController` w.r.t. the `TriggerInfo` that is passed down in `prepare`
As mentioned in https://github.com/bluesky/ophyd-async/pull/401#discussion_r1644527435 and https://github.com/bluesky/ophyd-async/pull/401#discussion_r1644506670, we need to decide what `StandardDetector` does with the `TriggerInfo` passed down to `prepare`. We should also consider #162 and #357 while we're at it. I propose:
- `async def DetectorController.arm(num, trigger, exposure) -> Status` -> `async def DetectorController.arm(trigger_info) -> None`, where the return value for `acquire.set(True)` is stored locally rather than returned
- Add a new `async def DetectorController.wait_for_disarmed(timeout=DEFAULT_TIMEOUT)` that waits for aquire to be False and does the successful completion check that will fail if the detector did anything apart from normal completion or manual disarming
- `async def DetectorController.disarm()` -> `async def DetectorController.start_disarm()` as it doesn't wait for completion

We should also document the various permutations of `TriggerInfo` and provide any common helpers like the one defined in #401. The cases we need to document:
- trigger=internal, livetime=deadtime=None: Internal trigger mode, with exposure and period untouched. The default for `trigger()` if `prepare()` not called
- trigger=any, livetime=float: Exposure set to livetime and period set to livetime+`get_deadtime(livetime)`
- trigger=any, livetime=float, deadtime=float: Exposure set to livetime and period set to deadtime
- num=0: Go until disarmed, if detector supports it
- num=+ve: Arm for this many frames

We still need to work out if we should provide things like min_exposure and max_framerate as well as `get_deadtime(exposure)`, but this will we tackled in a different issue

### PR DESCRIPTION
**Pull request by ZohebShaikh**
closes #405 #542 https://github.com/bluesky/ophyd-async/issues/549

### CODE REVIEW

File: src/ophyd_async/epics/adpilatus/_pilatus_controller.py [hunk 1]
```diff
@@ -59,8 +63,6 @@ async def arm(
             timeout=DEFAULT_TIMEOUT,
```
Comment (coretl) [inline src/ophyd_async/epics/adpilatus/_pilatus_controller.py:63]: This needs moving to `arm()`

**Review by coretl — CHANGES_REQUESTED** [review]

File: src/ophyd_async/epics/adkinetix/_kinetix_controller.py [hunk 2]
```diff
@@ -20,27 +21,31 @@ def __init__(
...
             DetectorTrigger.constant_gate,
         ]:
-            await self._drv.acquire_time.set(exposure)
-        return await adcore.start_acquiring_driver_and_ensure_status(self._drv)
+            await self._drv.acquire_time.set(trigger_info.livetime)
+
+    async def arm(self):
+        self._arm_status = await adcore.start_acquiring_driver_and_ensure_status(
+            self._drv
+        )
```
[the remark this answers is not in the document]
Comment (ZohebShaikh) [reply src/ophyd_async/epics/adkinetix/_kinetix_controller.py:44]: https://github.com/bluesky/ophyd-async/pull/568/commits/338f946769e65408bc2dc7a6430fee7548a43df0 did you mean something like this ?

File: src/ophyd_async/epics/adpilatus/_pilatus_controller.py [hunk 3]
```diff
@@ -47,21 +47,22 @@ async def prepare(self, trigger_info: TriggerInfo):
...
+        self._arm_status = await asyncio.gather(
+            adcore.start_acquiring_driver_and_ensure_status(self._drv),
+            # The pilatus has an additional PV that goes True when the camserver
+            # is actually ready. Should wait for that too or we risk dropping
+            # a frame
+            wait_for_value(
+                self._drv.armed,
+                True,
+                timeout=DEFAULT_TIMEOUT,
+            ),
```
Comment (coretl) [inline src/ophyd_async/epics/adpilatus/_pilatus_controller.py]: This was fine as it was...

File: src/ophyd_async/fastcs/panda/_control.py [hunk 4]
```diff
@@ -24,8 +26,9 @@ async def prepare(self, trigger_info: TriggerInfo):
         ), "Only constant_gate and variable_gate triggering is supported on the PandA"
 
     async def arm(self):
-        await asyncio.gather(self.pcap.arm.set(True))
-        await wait_for_value(self.pcap.active, True, timeout=1)
+        self._arm_status = await asyncio.gather(
+            self.pcap.arm.set(True), wait_for_value(self.pcap.active, True, timeout=1)
+        )
```
Comment (coretl) [inline src/ophyd_async/fastcs/panda/_control.py:31]: Same here

**Review by coretl — APPROVED** [review]

**Pull request merged**